pkg/simple/client/elasticsearch/esclient.go

```diff
@@ -60,12 +60,11 @@ type ElasticSearchClient struct {
 }
 
 func NewLoggingClient(options *ElasticSearchOptions) (*ElasticSearchClient, error) {
-	version := "6"
 	esClient := &ElasticSearchClient{}
 
 	if options.Version == "" {
 		var err error
-		version, err = detectVersionMajor(options.Host)
+		options.Version, err = detectVersionMajor(options.Host)
```

Comment:
not a good practice, since you've changed options value

Reply:
the same here, `options.Index = options.LogstashPrefix`. Should we refactor all of them?

Reply:
https://github.com/kubesphere/kubesphere/pull/799/files#diff-28320d9ea55aa094d44988b9161bbe6aR73-R77

Reply:
yes, please